contracts/OrderMixin.sol

```diff
@@ -141,60 +148,92 @@ abstract contract OrderMixin is
     /// Also allows to specify funds destination instead of `msg.sender`
     /// @param order Order quote to fill
     /// @param signature Signature to confirm quote ownership
-    /// @param makingAmount Making amount
-    /// @param takingAmount Taking amount
-    /// @param thresholdAmount Specifies maximum allowed takingAmount when takingAmount is zero, otherwise specifies minimum allowed makingAmount
+    /// @param orderAmounts Amounts to fill
     /// @param target Address that will receive swap funds
     /// @param permit Should consist of abiencoded token address and encoded `IERC20Permit.permit` call.
     /// @dev See tests for examples
     function fillOrderToWithPermit(
         Order memory order,
         bytes calldata signature,
-        uint256 makingAmount,
-        uint256 takingAmount,
-        uint256 thresholdAmount,
+        OrderAmounts calldata orderAmounts,
         address target,
         bytes calldata permit
     ) external returns(uint256 /* actualMakingAmount */, uint256 /* actualTakingAmount */) {
         require(permit.length >= 20, "LOP: permit length too low");
         (address token, bytes calldata permitData) = permit.decodeTargetAndData();
         _permit(token, permitData);
-        return fillOrderTo(order, signature, makingAmount, takingAmount, thresholdAmount, target);
+        return _fillOrderTo(order, signature, orderAmounts, target, new bytes(0));
     }
 
     /// @notice Same as `fillOrder` but allows to specify funds destination instead of `msg.sender`
     /// @param order Order quote to fill
     /// @param signature Signature to confirm quote ownership
-    /// @param makingAmount Making amount
-    /// @param takingAmount Taking amount
-    /// @param thresholdAmount Specifies maximum allowed takingAmount when takingAmount is zero, otherwise specifies minimum allowed makingAmount
+    /// @param orderAmounts Amounts to fill
     /// @param target Address that will receive swap funds
     function fillOrderTo(
         Order memory order,
         bytes calldata signature,
-        uint256 makingAmount,
-        uint256 takingAmount,
-        uint256 thresholdAmount,
+        OrderAmounts calldata orderAmounts,
         address target
     ) public returns(uint256 /* actualMakingAmount */, uint256 /* actualTakingAmount */) {
+        return _fillOrderTo(
+            order,
+            signature,
+            orderAmounts,
+            target,
+            new bytes(0)
+        );
+    }
+
+    /// @notice Same as `fillOrderTo` but allows for additional interaction between asset transfers
+    /// @param order Order quote to fill
+    /// @param signature Signature to confirm quote ownership
+    /// @param orderAmounts Amounts to fill
+    /// @param target Address that will receive swap funds
+    /// @param extraInteraction Optional interaction. If present, is triggered between asset transfers
+    function fillOrderToWithExtraInteraction(
+        Order memory order,
```

Comment:
the `order` struct is always immutable inside this file, so everywhere that we have `Order memory order` can be optimized to `Order calldata order` since its unmodified